### PR TITLE
[P4-1801] Create initial journey and view for Printing the Person Escort Record

### PR DESCRIPTION
**Pull request by teneightfive**
### What changed

This extends the move detail page to include the print journey once a PER has been confirmed.

It adds a new banner and an action to the view print route for a Person Escort Record.

**Note:** this doesn't include the additional elements needed for the "PER cover sheet" or the move detail summary yet.

### Why did it change

The PER needs to be printed out with the first release of this feature so this journey makes that clear
for users to do so.

### Issue tracking

- [P4-1801](https://dsdmoj.atlassian.net/browse/P4-1801)

### New banner on the move detail page

<kbd></kbd>

### Initial print view re-using existing summary elements

<kbd></kbd>

#### Automated testing

- [ ] Added unit tests to cover changes
- [ ] Added end-to-end tests to cover any journey changes

#### Manual testing

Has been tested in the following browsers:

- [x] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [x] No environment variables were added or changed

- [ ] Documented in the [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md)
- [ ] Added to [continous integration](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-book-secure-move-frontend)
- [ ] Added to staging environment (deployment repository)
- [ ] Added to preproduction environment (deployment repository)
- [ ] Added to production environment (deployment repository)

### Other considerations

Commit messages with a `fix` or `feat` type are automatically used to generate the [changelog](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/CHANGELOG.md) and
[GitHub release notes](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/releases) during the release task. Please make sure they will read well on their own in a
summary of changes and that the commit body gives a more detailed description of those changes if necessary.

- [x] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics
- [x] Update [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md) with any new instructions or tasks

### CODE REVIEW

File: app/person-escort-record/controllers/print-record.test.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+describe('Person Escort Record controllers', function () {
+  describe('#printRecord', function () {})
```
Comment (teneightfive) [inline app/person-escort-record/controllers/print-record.test.js:2]: Tests are still TODO...

**Comment by solidgoldpig** [comment]
Quick trivial spot

_P4-1801__Create_initial_journey_and_view_for_Printing_the_Person_Escort_Record_by_teneightfive_·_Pull_Request__703_·_ministryofjustice_hmpps-book-secure-move-frontend

Trailing comma when joining there

File: app/person-escort-record/controllers/print-record.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+const { filter, orderBy, sortBy } = require('lodash')
+
+const presenters = require('../../../common/presenters')
+
+function printRecord(req, res) {
+  const { framework, personEscortRecord, move } = req
```
Comment (solidgoldpig) [inline app/person-escort-record/controllers/print-record.js:6]:
```js
  const { framework, personEscortRecord, move = {} } = req
  const reference = move.reference
  const {profile = {}} = move
```
utterly trivial comment, but reads nicer to me

File: common/presenters/move-to-summary-list-component.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
+const { get, isNil } = require('lodash')
+
+const moveAgreedField = require('../../app/move/fields/move-agreed')
+const i18n = require('../../config/i18n')
+const filters = require('../../config/nunjucks/filters')
+
+function moveToSummaryListComponent(
```
[the remark this answers is not in the document]
Comment (solidgoldpig) [reply common/presenters/move-to-summary-list-component.js:7]: for once I think codeclimate has a point. Could we not extract the label munging?

[the remark this answers is not in the document]
Comment (teneightfive) [reply common/presenters/move-to-summary-list-component.js:7]: I just started down that path actually and realised this version doesn't actually need all the other details. We only need to show a subset of the information so going to remove what we don't need which should make it less samey...

**Comment by teneightfive** [comment]
> Trailing comma when joining there



> Quick trivial spot
> 
> P4-1801__Create_initial_journey_and_view_for_Printing_the_Person_Escort_Record_by_teneightfive·Pull_Request__703·_ministryofjustice_hmpps-book-secure-move-frontend
> 
> Trailing comma when joining there

@solidgoldpig Ah, do you have an ID that this is for? I wonder if that's actually just a keybashing typo because there is no join in that part of the display. Everything after the emdash comes from the textarea, same as the line above which has no trailing comma...

File: app/person-escort-record/controllers/print-record.js (hunk 2 above)
Comment (teneightfive) [reply app/person-escort-record/controllers/print-record.js:6]: I've tweaked this a little to support the record being accessed via a move or directly

**Comment by teneightfive** [comment]
Using admin to merge and overriding Code Climate as test coverage drop has been recorded in #711

**Pull request merged**